src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -952,7 +978,10 @@ UPDATE SET
         private SqlCommand BuildRenewLeasesCommand(SqlConnection connection, SqlTransaction transaction)
         {
             string matchCondition = string.Join(" OR ", this._rowMatchConditions.Take(this._rowsToProcess.Count));
-
+            if (string.IsNullOrEmpty(matchCondition))
+            {
+                this._logger.LogError($"MatchCondition resolved to empty with '{this._rowsToProcess.Count}' rowsToProcess.");
```

Comment:
@MaddyDev It wouldn't be a bad idea to just skip executing the query in this case as well - we know it's going to fail so no reason to add extra load to the server in that scenario.

If you'd rather not do that in this PR then I'd suggest filing a tracking issue for follow up

Reply:
Created https://github.com/Azure/azure-functions-sql-extension/issues/1108